plan/physical_plan_builder.go

```diff
@@ -1271,41 +1271,40 @@ func (p *Selection) makeScanController() *physicalPlanInfo {
 		}
 		child = ts
 	} else if p.controllerStatus == controlIndexScan {
-		var chosenPlan *PhysicalIndexScan
+		var (
+			chosenPlan                       *PhysicalIndexScan
+			bestEqualCount, bestInAndEqCount int
+		)
 		for _, idx := range indices {
 			condsBackUp := make([]expression.Expression, 0, len(corColConds))
 			for _, cond := range corColConds {
 				condsBackUp = append(condsBackUp, cond.Clone())
 			}
-			is := &PhysicalIndexScan{
-				Table:               ds.tableInfo,
-				Index:               idx,
-				Columns:             ds.Columns,
-				TableAsName:         ds.TableAsName,
-				OutOfOrder:          true,
-				DBName:              ds.DBName,
-				physicalTableSource: physicalTableSource{client: ds.ctx.GetClient()},
-			}
-			is.tp = TypeIdxScan
-			is.allocator = ds.allocator
-			is.SetSchema(ds.schema)
-			is.initIDAndContext(ds.ctx)
-			if is.ctx.Txn() != nil {
-				is.readOnly = p.ctx.Txn().IsReadOnly()
-			} else {
-				is.readOnly = true
-			}
-			var accessConds []expression.Expression
-			accessConds, _, is.accessEqualCount, is.accessInAndEqCount = DetachIndexScanConditions(condsBackUp, is.Index)
-			if len(accessConds) == 0 {
-				continue
-			}
-			better := chosenPlan == nil || chosenPlan.accessEqualCount < is.accessEqualCount
-			better = better || (chosenPlan.accessEqualCount == is.accessEqualCount && chosenPlan.accessInAndEqCount < is.accessInAndEqCount)
+			_, _, accessEqualCount, accessInAndEqCount := DetachIndexScanConditions(condsBackUp, idx)
+			better := chosenPlan == nil || bestEqualCount < accessEqualCount
+			better = better || (bestEqualCount == accessEqualCount && bestInAndEqCount < accessInAndEqCount)
```

Comment:
I think simply check access equal count is enough.
`better := accessEqualCount >= bestEqualCount`
Other wise, we need to add tests for `accesInAndEqCount`.